routes/home.js

```diff
@@ -27,8 +27,16 @@ router.get("/monitor", urlEncodedParser, handleIndexRoute);
 router.post("/scan", urlEncodedParser, async (req, res) => {
   const email = req.body.email;
   let foundBreaches;
+
   if (email) {
-    foundBreaches = await DBUtils.getBreachesForEmail(email);
+    if (req.session.scanResults && req.session.scanResults[email]) {
+      foundBreaches = req.session.scanResults[email];
+    } else {
+      req.session.scanResults = {};
```

Comment:
I'm a bit confused.

L32 says, if you have a `req.session.scanResults` object, and an `email` key is found, then do _X_.
Else, create an empty `req.session.scanResults` object.

Is it possible I'd have a `req.session.scanResults` object, but no match for that `email` key? It almost looks like if `req.session.scanResults` is truthy, but `req.session.scanResults[email]` is falsey, you'd erase the `req.session.scanResults` object.

Is it possible that you could have results for multiple email addresses in the `req.session.scanResults` object? Or is this strictly just caching results?

Reply:
I updated this code to not clear out the whole `req.session.scanResults` object when the submitted `email` isn't found.

We may have results for multiple email addresses, if a user submits multiple email addresses.